packages/next/client/page-loader.js

```diff
@@ -179,14 +202,17 @@ export default class PageLoader {
     ) {
       scriptRoute = scriptRoute.replace(/\.js$/, '.module.js')
     }
-
+    const url = isDependency
+      ? route
+      : `${this.assetPrefix}/_next/static/${encodeURIComponent(
+        this.buildId
+      )}/pages${scriptRoute}`
     if (
-      this.prefetchCache.has(scriptRoute) ||
+      document.querySelector(`link[rel="preload"][href^="${url}"]`) ||
```

Comment:
This change no longer accounts for the time in-between script tag insertion and download/execution to register `__NEXT_PAGE__${route}`.

Previously, the set would prevent double pre-fetching if this was called while the script was loading.

Note, this only happens on older browsers that don't support `link[rel="preload"]` and we fall back to a `<script>` element.

Reply:
Actually, `loadPage` already handles this via `this.loadingRoutes`. 💅